### PR TITLE
Templates => Auth 2.0 

### PR DESCRIPTION
**Pull request by HaoK**
Initial PR for adding Auth 2.0, I also fixed the change to Configure[App]Configuration (cc @glennc)

I only verified that I could compile and run Individual/Single/MultiOrg.

@tratcher @eilon @mlorbetske @danroth27 @blowdart @ajcvickers

### CODE REVIEW

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Controllers/AccountController.cs [hunk 1]
```diff
@@ -129,7 +126,7 @@ public IActionResult SignedOut()
         public async Task<IActionResult> Login(string returnUrl = null)
         {
             // Clear the existing external cookie to ensure a clean login process
-            await HttpContext.Authentication.SignOutAsync(_externalCookieScheme);
+            await HttpContext.Authentication.SignOutAsync(IdentityCookieOptions.ExternalScheme);
```
Comment (Tratcher) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Controllers/AccountController.cs]: -Authentication?

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Views/Account/Login.cshtml [hunk 2]
```diff
@@ -59,7 +60,8 @@
             <h4>Use another service to log in.</h4>
             <hr />
             @{
-                var loginProviders = SignInManager.GetExternalAuthenticationSchemes().ToList();
+                var schemes = await SchemeProvider.GetAllSchemesAsync();
```
Comment (Tratcher) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Views/Account/Login.cshtml]: Start by adding a manual filter for the identity cookies?

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Views/Account/Login.cshtml [hunk 3]
```diff
@@ -1,8 +1,9 @@
 ﻿@using System.Collections.Generic
 @using Microsoft.AspNetCore.Http
-@using Microsoft.AspNetCore.Http.Authentication
+@using Microsoft.AspNetCore.Authentication
 @model LoginViewModel
 @inject SignInManager<ApplicationUser> SignInManager
```
Comment (Eilon) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Views/Account/Login.cshtml]: Looks like `SignInManager` isn't used anymore - should it be removed?

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Views/Account/Login.cshtml (hunk 2 above)
Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Views/Account/Login.cshtml]: Maybe we can start by just adding a property to the AuthenticationScheme called Remote, and only set that for our Remote handlers (OAuth based + Twitter), then we can filter for Remote schemes?

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Views/Account/Login.cshtml (hunk 3 above)
Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Views/Account/Login.cshtml]: Yup, will remove

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Controllers/AccountController.cs (hunk 1 above)
Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Controllers/AccountController.cs]: Yup, I'll remove

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Views/Account/Login.cshtml (hunk 2 above)
Comment (Tratcher) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Views/Account/Login.cshtml]: You're baking template logic into the framework.

Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Views/Account/Login.cshtml]: Well we need something, I guess for now, lets just bring back what we had, I'll add DisplayName to AuthenticationScheme and have the Auth's set it like before, and resurrect the SignInManager API for Preview 1, and we can figure out a longer term plan for Preview 2 if need

Comment (Tratcher) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Views/Account/Login.cshtml]: That's a lot more work than adding a temp filter right here.

Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Views/Account/Login.cshtml]: Not really, you underestimate how many places I had to do this exact code pattern last night :)

**Comment by HaoK** [comment]
Updated with the new iteration, SignInManager.GetExternal... is back, and DisplayName has been restored.

**Comment by HaoK** [comment]
Okay I think changes are pretty much good to go, @danroth27 and I are doing final runtime verification, look ok to everyone to merge today?  cc @DamianEdwards @davidfowl

**Pull request merged**